### ISSUE TITLE
Loading pre-trained/warmstarted policies for fine-tuning

### ISSUE THREAD

**Issue by yawen-d** [HumanCompatibleAI/imitation#376]
At some point, we'll need a feature for loading a pre-trained/warmstarted policy for continuous training. 

This feature is currently required in [`imitation.scripts.train_preference_comparisons`](https://github.com/HumanCompatibleAI/imitation/blob/dacb2425e9d19d57e318578a414b50cb98ead647/src/imitation/scripts/train_preference_comparisons.py#L101), and will be required in `imitation.scripts.train_rl`.

To load an existing policy for further training, we at least need to have the following features:
- enabling the setting of `vec_normalize.training=True`.
- No need for introducing a NormalizePolicy like what's done in [`serialize`](https://github.com/HumanCompatibleAI/imitation/blob/dacb2425e9d19d57e318578a414b50cb98ead647/src/imitation/policies/serialize.py#L168).

Also, there are ways to simplify `imitation.policies.serialize`:
- We could refactor policy to algorithm where appropriate. The naming of `policy` and `algorithm` in `imitation.policies.serialize` is kind of ambiguous. e.g. in [`load_policy()`](https://github.com/HumanCompatibleAI/imitation/blob/dacb2425e9d19d57e318578a414b50cb98ead647/src/imitation/policies/serialize.py#L210), "ppo" is regarded as one candidate of policy types, while PPO is an `OnPolicyAlgorithm` in SB3.
- Policy loading should be independent with choices of algorithms. e.g. SAC and PPO could load the same policy. The current [method](https://github.com/HumanCompatibleAI/imitation/blob/dacb2425e9d19d57e318578a414b50cb98ead647/src/imitation/policies/serialize.py#L197) use the class method of PPO and then extract its policy attribute. But the class method was defined in [BaseAlgorithm](https://stable-baselines3.readthedocs.io/en/master/_modules/stable_baselines3/common/base_class.html#BaseAlgorithm.load). It seems fine to get rid of additional specifiers of algorithm choice.

The above more or less serves as some notes for me. Happy to give it a go.

**Comment by AdamGleave** [comment]
Checking in on this, I think we now have support for this feature in `train_rl` and `train_preference_comparisons` via the `agent_path` config parameter. But not in `train_imitation` or `train_adversarial`. Does that seem right to you, @yawen-d?

The `serialize` simplification I think should be its own issue. I think using `BaseAlgorithm.load` won't work as the first parameter it takes is `cls` which is diferent depending on whether you call e.g. `DQN.load` or `PPO.load`. Even if it did work the SB3 [examples](https://stable-baselines3.readthedocs.io/en/master/guide/examples.html) do use the algorithm's load method e.g. `DQN.load` so I think we might potentially get broken in the future. 

Separating loading policy and algorithms is an interesting idea and might be worth pursuing, at least if we have a research use case for that (e.g. train policy with one algorithm, fine-tune it with a different one).

**Comment by yawen-d** [comment]
> Checking in on this, I think we now have support for this feature in train_rl and train_preference_comparisons via the agent_path config parameter. But not in train_imitation or train_adversarial. Does that seem right to you, @yawen-d?

Yep, that's correct.

> The serialize simplification I think should be its own issue. I think using BaseAlgorithm.load won't work as the first parameter it takes is cls which is diferent depending on whether you call e.g. DQN.load or PPO.load. Even if it did work the SB3 [examples](https://stable-baselines3.readthedocs.io/en/master/guide/examples.html) do use the algorithm's load method e.g. DQN.load so I think we might potentially get broken in the future.

After looking more into this, I think it's not possible to use DQN policy to load a PPO policy because some policies are customized for the algorithms. e.g. DQN uses [`DQNPolicy`](https://github.com/DLR-RM/stable-baselines3/blob/ef10189d80dbb2efb3b5391cba4eb3c2ab5c7aae/stable_baselines3/dqn/policies.py#L208), PPO uses [`ActorCriticPolicy`](https://github.com/DLR-RM/stable-baselines3/blob/ef10189d80dbb2efb3b5391cba4eb3c2ab5c7aae/stable_baselines3/ppo/policies.py#L5), and SAC uses a customized [SACPolicy](https://github.com/DLR-RM/stable-baselines3/blob/ef10189d80dbb2efb3b5391cba4eb3c2ab5c7aae/stable_baselines3/sac/policies.py#L188). I agree that we don't need to pursue separating loading policy and algorithms, unless we have a research use case for that.

I would think it's still useful to load a pretrained policy for `train_adversarial`, but I'm inclined to closing this issue for now until we have a use case for train_adversarial to load a pretrained policy.

**Comment by AdamGleave** [comment]
Thanks for clarifying this Yawen!

Personally I'd like to see support for loading pretrained policy in `train_adversarial` and `train_imitation`. I think we will have research use cases in the future, and it seems nice to have consistency here.
